embedded-graphics/src/primitives/polyline.rs

```diff
@@ -0,0 +1,358 @@
+        if vertices.len() < 2 {
+            panic!("Polyline must contain at least two vertices");
+        }
+
+        Self {
+            vertices,
+            translate: Point::zero(),
+        }
+    }
+}
+
+impl<'a> Primitive for Polyline<'a> {
+    type PointsIter = PolylineIterator<'a>;
+
+    fn points(&self) -> Self::PointsIter {
+        self.into_iter()
+    }
+}
+
+impl<'a> Dimensions for Polyline<'a> {
+    fn bounding_box(&self) -> Rectangle {
+        let top_left = self
+            .vertices
+            .iter()
+            .map(|v| *v + self.translate)
+            .fold(Point::new(core::i32::MAX, core::i32::MAX), |accum, v| {
+                Point::new(accum.x.min(v.x), accum.y.min(v.y))
+            });
+
+        let bottom_right = self
+            .vertices
+            .iter()
+            .map(|v| *v + self.translate)
+            .fold(Point::new(core::i32::MIN, core::i32::MIN), |accum, v| {
+                Point::new(accum.x.max(v.x), accum.y.max(v.y))
+            });
+
+        Rectangle::with_corners(top_left, bottom_right)
+    }
+}
+
+impl<'a> Transform for Polyline<'a> {
+    /// Translate the polyline from its current position to a new position by (x, y) pixels, returning
+    /// a new `Polyline`. For a mutating transform, see `translate_mut`.
+    ///
+    /// ```
+    /// # use embedded_graphics::primitives::Polyline;
+    /// # use embedded_graphics::prelude::*;
+    /// let points = [
+    ///     Point::new(5, 10),
+    ///     Point::new(7, 7),
+    ///     Point::new(5, 8),
+    ///     Point::new(10, 10),
+    /// ];
+    ///
+    /// let polyline = Polyline::new(&points);
+    /// let moved = polyline.translate(Point::new(10, 12));
+    ///
+    /// assert_eq!(polyline.bounding_box().top_left, Point::new(5, 7));
+    /// assert_eq!(moved.bounding_box().top_left, Point::new(15, 19));
+    /// ```
+    fn translate(&self, by: Point) -> Self {
+        Self {
+            translate: by,
+            ..*self
+        }
+    }
+
+    /// Translate the polyline from its current position to a new position by (x, y) pixels.
+    ///
+    /// ```
+    /// # use embedded_graphics::primitives::Polyline;
+    /// # use embedded_graphics::prelude::*;
+    /// let points = [
+    ///     Point::new(5, 10),
+    ///     Point::new(7, 7),
+    ///     Point::new(5, 8),
+    ///     Point::new(10, 10),
+    /// ];
+    ///
+    /// let mut polyline = Polyline::new(&points);
+    ///
+    /// polyline.translate_mut(Point::new(10, 12));
+    ///
+    /// assert_eq!(polyline.bounding_box().top_left, Point::new(15, 19));
+    /// ```
+    fn translate_mut(&mut self, by: Point) -> &mut Self {
+        self.translate += by;
+
+        self
+    }
+}
+
+/// An iterator over all pixel positions on the polyline
+#[derive(Copy, Clone, Eq, PartialEq, Ord, PartialOrd, Hash, Debug)]
+pub struct PolylineIterator<'a> {
+    stop: bool,
+    vertices: &'a [Point],
+    translate: Point,
+    segment_iter: ThickLineIterator,
```

Comment:
This should use `line::Points` instead of `ThickLineIterator`. They are equivalent at the moment, but `Points` will be optimized at some point (#314).